src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -340,7 +365,78 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
         }
     }
 
-    private long writePQ(SequentialWriter writer) throws IOException
+    private boolean isFastFindRows(BiMap<Integer, Integer> ordinalMap)
+    {
+        boolean canFastFindRows;
+        canFastFindRows = true;
+        int minRow = Integer.MAX_VALUE;
+        int maxRow = Integer.MIN_VALUE;
+        for (VectorPostings<T> vectorPostings : postingsMap.values())
+        {
+            if (vectorPostings.getRowIds().size() != 1)
+            {
+                canFastFindRows = false;
+                ordinalMap.clear();
```

Comment:
I don't think we need to pass in the map?  Much cleaner if we instantiate and build it all  locally.  then return an empty Map if the fast-path turns out to not work